cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueAdmin.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.transaction.queue;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.queue.QueueName;
+import co.cask.cdap.data2.datafabric.DefaultDatasetNamespace;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.proto.Id;
+
+/**
+ * Common implementation of table-based QueueAdmin
+ */
+public abstract class AbstractQueueAdmin implements QueueAdmin {
+  private final String unqualifiedTableNamePrefix;
+  private final String unqualifiedConfigTableNameSuffix;
+  protected final DefaultDatasetNamespace namespace;
+
+  public AbstractQueueAdmin(CConfiguration conf, QueueConstants.QueueType type) {
+    // system scoped
+    // todo: we have to do that because queues do not follow dataset semantic fully (yet)
+    this.unqualifiedTableNamePrefix = Constants.SYSTEM_NAMESPACE + "." + type.toString();
+    this.unqualifiedConfigTableNameSuffix = Constants.SYSTEM_NAMESPACE + "." + QueueConstants.QUEUE_CONFIG_TABLE_NAME;
+    this.namespace = new DefaultDatasetNamespace(conf);
+  }
+
+  /**
+   * @param queueTableName actual queue table name
+   * @return namespace id that this queue belongs to
+   */
+  public static String getNamespaceId(String queueTableName) {
```

Comment:
The way we construct the name (TableId object will be used) will have to change when we integrate with hbase namespaces, and so I'll postpone it to then. (otherwise I might do work which wouldn't be tossed at that stage).
https://issues.cask.co/browse/CDAP-1594